Correct handling of dtend and duration for vevents

sunesimonsen opened the pull request:
Corresponding to https://tools.ietf.org/html/rfc5545#page-54 it is
allowed to supply a duration instead of a dtend and leave both of them
out. If there is only a dtstart and it is a date, it should be treated
an all day event. If there is only a dtstart and it is a time it
indicates that point in time.

This pull request fixes all the cases.

kewisch commented on build/ical.js:
```diff
@@ -6526,6 +6536,11 @@ ICAL.Event = (function() {
+      if (duration) {
+          return duration;
+      }
+
```
Could you flip this around so it matches the flow that get endDate has, i.e.

``` javascript
if (!duration) {
  return this.endDate.subtractDate(this.startDate);
}

return duration;
```

kewisch commented on build/ical.js:
```diff
@@ -6518,7 +6518,17 @@ ICAL.Event = (function() {
+          if (duration) {
+              endDate.addDuration(duration);
+          } else if (endDate.isDate) {
+              endDate.addDuration(new ICAL.Duration({ days: 1 }));
```
Instead of constructing a new `ICAL.Duration` just for the purpose of incrementing the day, could you just increment using the `day` property?

kewisch commented on samples/duration_instead_of_dtend.ics:
```diff
@@ -0,0 +1,39 @@
+PRODID:-//Google Inc//Google Calendar 70.9054//EN
+VERSION:2.0
+CALSCALE:GREGORIAN
+X-WR-CALNAME:calmozilla1@gmail.com
```
Is this one of your addresses? If not, please use something at example.com to avoid adding real addresses in the source.

sunesimonsen replied:
Sorry about that, I did just copy the content from samples/minimal.ics

kewisch commented on samples/duration_instead_of_dtend.ics:
```diff
@@ -0,0 +1,39 @@
+DTSTART;TZID=America/Los_Angeles:20120630T060000
+DURATION:P1D
+DTSTAMP:20120724T212411Z
+UID:dn4vrfmfn5p05roahsopg57h48@google.com
```
Same here, I think its better to use example.com

kewisch commented on samples/only_dtstart_date.ics:
```diff
@@ -0,0 +1,38 @@
+PRODID:-//Google Inc//Google Calendar 70.9054//EN
+VERSION:2.0
+CALSCALE:GREGORIAN
+X-WR-CALNAME:calmozilla1@gmail.com
```
Same comments on domain here and in all other samples.

kewisch commented:
Sorry for the delay. r=me with above changes.

sunesimonsen commented:
I'll fix the problems and get back to you later today.

sunesimonsen committed:
Correct handling of dtend and duration for vevents

Corresponding to https://tools.ietf.org/html/rfc5545#page-54 it is
allowed to supply a duration instead of a dtend and leave both of them
out. If there is only a dtstart and it is a date, it should be treated
an all day event. If there is only a dtstart and it is a time it
indicates that point in time.

sunesimonsen committed:
Build package

sunesimonsen commented:
@kewisch I fixed the problems you highlighted and force pushed the branch.

sunesimonsen commented:
Thanks
